### ISSUE TITLE
Freezing upon loading savestates

### ISSUE THREAD

**Issue by BLCRAFT210** [Felk/dolphin#32]
I've noticed a few ways that savestate functions can cause Dolphin to freeze. The first is when you attempt to load a savestate with the scripting API that wasn't created using the API. (Edit: This actually works, but for some reason the game speed is much faster than normal upon loading.) Likewise, attempting to load a savestate with a hotkey will also cause a freeze if the savestate was created using the API. Finally, loading a savestate twice in the same program causes a freeze. This is a quick program that causes that:
```
from dolphin import event, savestate

await event.frameadvance()
savestate.load_from_file([path to file]) #this one works fine
for i in range(60):
    await event.frameadvance()
savestate.load_from_file([path to file]) #this one crashes
```
I don't know why these freezes happen, and there might be some other bugs with the savestate module.

**Comment by BLCRAFT210** [comment]
Actually, I just tested using a different savestate and it worked just fine. I don't know why it works in one case but not the other.

**Comment by BLCRAFT210** [comment]
> Actually, I just tested the above program using a different savestate and it worked just fine. I don't know why it works with one savestate but not the other.

I found some potential clues to what causes these freezes. The savestate I mention here freezes on the second load when V-Sync is enabled, but it works fine again when V-Sync is disabled. I then found an easier fix which worked for all savestates I tested: setting the emulation speed limit to 10%.

I don't know much about how Dolphin or this branch works, but I have some guesses for what's going on here. I think the problem isn't with loading a savestate twice; it's with loading a savestate in the middle of emulation. It appears that somehow, giving Dolphin more time to fully load a savestate before it advances to the next frame fixes this problem. Therefore, this freeze may be caused by Dolphin trying to step to the next frame while `savestate.load_from_file` is in the middle of loading a savestate. Perhaps the Python function is missing something that Dolphin's savestate implementation does to prevent this problem.

**Comment by BLCRAFT210** [comment]
Also, as @VIPTankz found out, adding `time.sleep(1)` before loading the savestate fixes this.

**Comment by BLCRAFT210** [comment]
@JimB16 what's an example of a program that loads savestates properly?

**Comment by Felk** [comment]
Yeah, it's possible there's a bunch of issues wrt race conditions. I'll have to make a stresstest and see if more rigorous synchronization between python code and emulation will fix it. Thanks for bringing it up, I'll leave this issue open for now.

**Comment by benjaminjmiddleton** [comment]
I'm also experiencing the same issue, only none of the workarounds here seem to fix it, disabling VSync, making new save states, slowing emulator speed, using time.sleep() before or after loading. I haven't been able to successfully load a state the "second time" in a script once. For example, this simple script always fails:

```
savestate.load_from_slot(1) # works fine

await event.frameadvance()

savestate.load_from_slot(1) # never works
```

So, bump

**Comment by benjaminjmiddleton** [comment]
Thanks, I didn't know about that setting! I will try that when I get the chance later this week.
